Source/genetic_factory.cpp

```diff
@@ -1,21 +1,79 @@
 #include "genetic_factory.h"
 
-chromosome* genetic_factory::create_chromosome(chromosome_metadata* chromosome_metadata) {
-	return nullptr;
+genetic_factory& genetic_factory::get_instance() {
+    static genetic_factory instance;
+    return instance;
+}
+
+//region plugin registration
+
+void genetic_factory::register_selection_algorithm(const try_create_selection_algorithm_t try_create) {
```

Comment:
There are so many different variants of `register_selection_algorithm`. I'm having a hard time keeping them all straight.

Reply:
This is the genetic factory version of 'register_selection_algorithm' function. This is the one that will actually put the 'registered' constructor into a std::vector (read list) in the core assembly's memory.